packer/cleanup-ami-and-snapshots.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+import boto3
+from botocore.exceptions import ClientError
+
+REGIONS = [
+    "us-west-2",
+    "eu-central-1",
+    "ap-southeast-1",
+    "eu-west-2"
+]
+
+EPOCH_IMAGE_NAME = "epoch-ubuntu-16.04"
+
+def get_account_id():
+    sts = boto3.client("sts")
+    return sts.get_caller_identity()["Account"]
+
+def get_stale_amis(ec2_client, epoch_image_name):
+    amis = []
+    images = ec2_client.describe_images(
+        Filters = [
+            {
+                "Name": "name",
+                "Values": [
+                epoch_image_name + "*"
+                ]
+            }
+        ],
+        Owners = [
+            get_account_id()
+        ]
+    )
+    used_amis = get_used_amis(ec2_client)
+
+    for image in images["Images"]:
+        if image["ImageId"] not in used_amis:
+            snaps = []
+            for block in image["BlockDeviceMappings"]:
+                if "Ebs" in block:
+                    snaps.append(block["Ebs"]["SnapshotId"])
+            amis.append({"ImageId": image["ImageId"],"CreationDate": image["CreationDate"],"Snapshots": snaps})
+
+    amis.sort(key=lambda image: image["CreationDate"], reverse = True)
+
+    return amis[3:]
+
+def get_used_amis(ec2_client):
+    used_amis = []
+    response = ec2_client.describe_instances()
+
+    for reservation in response["Reservations"]:
+        for instance in reservation["Instances"]:
+            used_amis.append(instance["ImageId"])
+    return list(set(used_amis))
+
+def deregister(ec2_client, ids):
+    for i in ids:
+        print(i)
+        try:
+            ec2_client.deregister_image(
+                ImageId = i["ImageId"]
+            )
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'InvalidAMIID.Unavailable':
+                print("Already deleted")
+        try:
```

Comment:
this block should be removed if you want to handle it in the "main" and exit with status 1